### PR TITLE
Hardhat plugin

### LINKED ISSUE #121: Create `@zk-groups/hardhat` plugin
**Description**:

An Hardhat plugin can be useful when developers need to deploy their own `ZKGroups.sol` contract locally to test their contracts.

**Tasks**:

- [x] Create NX lib
- [x] Publish NPM package

### PR DESCRIPTION
**Pull request by cedoor**
## Description

This PR creates a new NX library which can be used as a Hardhat plugin to deploy the current ZKGroups contracts.

## Related Issue

Closes #121 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: libs/hardhat/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import "hardhat-dependency-compiler"
+import "@nomiclabs/hardhat-ethers"
+import "./tasks/deploy-zk-groups"
+import "./tasks/deploy-zk-groups-semaphore"
+
+extendConfig(
+    (config: HardhatConfig, userConfig: Readonly<HardhatUserConfig>) => {
+        config.dependencyCompiler.paths = [
+            "@semaphore-protocol/contracts/base/Pairing.sol",
```
Comment (saleel) [inline libs/hardhat/src/index.ts:12]: I haven't tested this, but I think we need to point to the ZkGroups solidity files here

Comment (cedoor) [reply libs/hardhat/src/index.ts:12]: Yes, I think we need `"@zk-groups/contracts/ZKGroups.sol"`. Let me know as soon as you test it

File: libs/hardhat/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "@zk-groups/hardhat",
-    "type": "module",
     "version": "0.1.1",
+    "main": "src/index.ts",
```
Comment (cedoor) [inline libs/hardhat/package.json:4]: @saleel This should automatically be added by NX.

It generates:

```json
  "main": "./src/index.js",
  "typings": "./src/index.d.ts"
```

Is it correct?

Comment (saleel) [reply libs/hardhat/package.json:4]: But we dont have a `index.js` file in src 🤔

**Pull request merged**